### DISCUSSION TITLE
Costum Sliders with AbstractPlutoDingetjes.jl still fail to fetch possible values

### DISCUSSION THREAD

**Discussion by mgraeser** [JuliaPluto/PlutoSliderServer.jl#125] [Issues] [answered]
Hi Everyone,

I would like to have a HTML Slider which is a bit wider than the standard one in oder to adjust it in finer steps and make it easier for the user to find the right spot. For this I added a Custum Slider using the example in AbstractPlutoDingetjes.jl and added the style='width:100%;' tag. 

In the precompilation export it states that it can not extract the set of possible values eventhough I used the AbtractPlutoDingetjes.jl package. 
Any suggestions how to get a wide slider running with the precompilation package?
Thank you for your help
Matthias

```
struct MySlider
    range::AbstractRange{<:Real}
end

Base.show(io::IO, m::MIME"text/html", s::MySlider) = show(io, m, HTML("<input type=range min=$(first(s.range)) step=$(step(s.range)) max=$(last(s.range)) style='width: 100%;'>"))

AbstractPlutoDingetjes.Bonds.initial_value(s::MySlider) = first(s.range)

Base.get(s::MySlider) = first(s.range)
end
```

![image](https://github.com/JuliaPluto/PlutoSliderServer.jl/assets/24937954/69b72d91-9824-4c7c-89ca-61e32c2f2834)

**Comment by Pangoraw** [comment, marked as answer]
You need to implement [`AbstractPlutoDingetjes.Bonds.possible_bond_values(::MySlider)`](https://github.com/JuliaPluto/AbstractPlutoDingetjes.jl/blob/2858a9d0c81c98ec5d9a627fbbba49ceb72dea0e/src/AbstractPlutoDingetjes.jl#L175-L206) in order to be able to precompute all possible values.
See for example, the implementation for `AbstractPlutoDingetjes.Bonds.possible_bond_values(::PlutoUI.Slider)` [here](https://github.com/JuliaPluto/PlutoUI.jl/blob/1b06470a5ae951c0e0845a22d8879843fecbccc9/src/Builtins.jl#L209).
